stream-tests/src/test/scala/org/apache/pekko/stream/scaladsl/FlowExistsSpec.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.pekko.stream.scaladsl
+
+import org.apache.pekko.stream.testkit._
+import org.apache.pekko.stream.testkit.scaladsl._
+import org.scalatest.concurrent.ScalaFutures
+
+class FlowExistsSpec extends StreamSpec with ScalaFutures {
+
+  "An Exists" must {
+
+    def greaterThan3(i: Int) = i > 3
+
+    "emit true for empty Source" in {
```

Comment:
false？

Reply:
After talking to fs2 people on discord, it seems like that fs2 returns false was a historical mistake of implementation restriction.
So yes, we should change our implementation to return false for empty stream.